sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -317,21 +317,43 @@ private[aggregate] object CollectTopK {
 case class ListAgg(
     child: Expression,
     delimiter: Expression = Literal(null),
-    orderExpressions: Seq[SortOrder] = Nil,
+    orderChildExpressions: Seq[Expression] = Nil,
+    orderDirections: Seq[SortDirection] = Nil,
+    orderNullOrderings: Seq[NullOrdering] = Nil,
+    orderSameOrderExpressions: Seq[Seq[Expression]] = Nil,
```

Comment:
do we need this? There is no `SortOrder` anymore, and we don't need it to make the planner sort to remove duplicated sorts.